### DISCUSSION TITLE
LIKE / wildcard search

### DISCUSSION THREAD

**Discussion by ravisharmaa** [meilisearch/product#356] [Feedback & Feature Proposal]
**I am so far loving the mielisearch but the missing wildcard search is frustrating me**
I have a recipe data set which has taste index which can contain fields like

```json
[
  {
    "name": "Tuna",
    "taste": "Sweet,Sour" 
  },
  {
    "name": "Turkey",
    "taste": "Sweet" 
  },
  {
    "name": "Pistachio",
    "taste": "Sour" 
  }
]

```

I added the `searchableAttribute` to be the `name` property and the `filterableAttribute` to be the `taste` property. Now when I send the query as 

```json
{
    "q": "Tu",
    "filter":"(taste LIKE '%Sw%')"
}

```
And want both records `Tuna` and `Turkey`. I can't find any documentation to accomplish this. Can anybody help please!!

Thank you.

**Comment by curquiza** [comment]
Hello @ravisharmaa 

Thanks for your feedback

Since MeiliSearch cannot handle the `LIKE` search, this is a feature request I moved to the right repository, so that we ensure your request is taken into account by our product team!

**Reply by ravisharmaa** [reply]
@curquiza 
I worked this out anyone wishing to accomplish this might leverage the faceted search pattern to do so. Meaning, they might need to search on a different data structure altogether. For example, I wanted to search on `Tuna` and `Turkey` using an approach used in the database, which does not work quite well with search engines.  As, the MeiliSearch and other search services do not provide like search and provide either Tuna or Turkey but not both.

```json
[
  {
    "name": "Tuna",
    "taste": "Sweet,Sour" 
  },
  {
    "name": "Turkey",
    "taste": "Sweet" 
  },
  {
    "name": "Pistachio",
    "taste": "Sour" 
  }
]

```


However, a slight change in the data-structure which could help resolve issues such as 


```json

[
  {
    "name": "Tuna",
    "tags": [
        "Sweet",
         "Sour"
      ]
  },
  {
    "name": "Turkey",
    "tags": [
       "Sweet"
   ] 
  },
  {
    "name": "Pistachio",
    "tags": [
        "Sour"
    ] 
  }
]


```

Now with the help of facetted search available within Meilisearch. Records can be search such as,

```json
{
   "q": "T",
    "filter":"tags = 'Sweet' or tags = 'Sour'"
}

```

which worked in my situation. 
Thanks

**Reply by gmourier** [reply]
Thanks @ravisharmaa !

**Reply by ibrahimjamil** [reply]
Not a good option as when we want to search through paragraph we could do list of words here 
its not generic in any case

**Reply by ibrahimjamil** [reply]
usecase
if you want to search word in paragraph then it could not be sufficient to convert all words to list 
and then seach for we could have simple sql like syntax here to search with like operator
keyword Like `%{wordtosearch}%`

**Comment by MathiasWeisheit** [comment]
Hi, how can I achive to query "rkey" and find the record wirth "Turkey"

**Reply by gmourier** [reply]
Hi @MathiasWeisheit, 

This is not currently possible. Meilisearch only searches by the prefix.

How important is it for you to be able to do this and why?

Thanks ✨

**Reply by MathiasWeisheit** [reply]
For us it is very important. Currently I have a workaround. I search via database like search and also with scout and meili. But this is not a solution. What do you think?

**Reply by MathiasWeisheit** [reply]
My cusomer whats to search for parts of a word. Like: 
consultant must find businessconsultant

**Reply by irevoire** [reply]
And I guess you can’t create in advance a list of synonyms doing the association `businessconsultant -> consultant`?

**Reply by MathiasWeisheit** [reply]
yes this is possible. But the content is created by many users. So for every failed search I have to create synonyms. ... I think this will not be the best solution. 
Wikipedia also does not find "Unternehmensberater" with searching "Berater" https://de.wikipedia.org/w/index.php?search=berater&title=Spezial:Suche&profile=advanced&fulltext=1&ns0=1

Google also not: https://www.google.com/search?q=wiki+ernehmensberater&sxsrf=ALiCzsZW-MKGTo6IH8EYSqThnBjAP8Ziaw%3A1661854746892&ei=GuQNY_H8Nd2qxc8P4qC92As&ved=0ahUKEwjx8eyuq-75AhVdVfEDHWJQD7sQ4dUDCA0&uact=5&oq=wiki+ernehmensberater&gs_lcp=Cgdnd3Mtd2l6EAM6BwgAEEcQsAM6BwgjELACECc6CAgAEB4QCBANSgQIQRgASgQIRhgAUO0FWMwLYMgPaAFwAXgAgAGnAYgBlgOSAQMwLjOYAQCgAQHIAQi4AQPAAQE&sclient=gws-wiz

I think every search engine has this problem?
Perhaps Meili Search solves this issue :-)

**Comment by polhemcom** [comment]
I would also like this feature - is this feature implemented yet or still talked being discussed?

**Reply by macraig** [reply]
Hi @joelbacklund , we haven't implemented the feature yet but we're happy to collect feedback and discuss how it could be implemented in the future. Would you mind sharing your use case and how you're working around this limitation for now (if that's the case)?

**Comment by lockness-Ko** [comment]
Thought I'd jump in to try and get some support for this :)

I'm looking into using meilisearch for security/threat hunting purposes due to it being lightweight but it seems to be limited by the lack of wildcard search.

Something simple like being able to do:
```json
{
  "filter": [
    "field = 'something*'",
    "field2 = 'some*thing'",
    "field3 = '*something'"
  ]
}
```

Or, at least just prefix and suffix searches would help a lot 😄

**Reply by macraig** [reply]
Hi @lockness-Ko , thanks for your feedback. We've [recently released a CONTAINS filter operator](https://github.com/orgs/meilisearch/discussions/544#discussioncomment-10193879), and are planning to work on the STARTS_WITH operator next.

**Reply by lockness-Ko** [reply]
Awesome 😎 thank you
